brave/browser-laptop
# Splits sites object in the app state

Submitter Checklist:


Resolves #1646 (Wrong favicon is shown for bookmark)
Resolves #1856 (Removing sites from history - `siteUtil` functionality)
Resolves #2655 (Bookmark name of a file does not appear in the autocomplete list)
Resolves #2771 (Bookmark folder is created left of bookmark item, not right)
Resolves #3646 (Bookmarks title changes when navigating around the page)
Resolves #3694 (Add automated test for making sure bookmark custom title appears on bookmark toolbar)
Resolves #4224 (Hidden history entries appear by deleting the latest entry)
Resolves #4260 (Edited information of bookmark does not appear on bookmark dialog via star icon)
Resolves #4833 (Bookmark title dynamically changed)
Resolves #4868 (Rebookmarking from context menu does not reset bookmark title)
Resolves #4929 (Bookmark title automatically changes on the bookmark form)
Resolves #5072 (about:history omits recent history)
Resolves #5382 (clicking entries on about:history is laggy)
Resolves #5699 (Last Visited on Bookmark Manager is not recorded for newly created bookmarks)
Resolves #6104 (Address bar should autocomplete from custom bookmark titles)
Resolves #6108 (Bookmark title on bookmark toolbar is renamed)
Resolves #6585 (Remove customTitle from siteDetail object)
Resolves #8022 (Bookmark moved to a folder on the toolbar cannot be removed via the star button)
Resolves #9301 (Changing bookmark location causes duplicate bookmarks)
Resolves #9326 (Custom bookmark title is not being displayed in suggestions) 
Resolves #9978 (Split sites into separate entities)
Resolves #10026 (Bookmarks toolbar : imported sorted bookmarks are represented in wacky order)

Auditors: @bbondy @ayumi @diracdeltas @bsclifton @darkdh 

Test Plan:
Run general regression tests first: https://github.com/brave/browser-laptop/wiki/Manual-Tests

1) Bookmark
- try to add a bookmark
2) Bookmark
- try to add a bookmark, but into the second level folder
3) Bookmark folder
- try to add a bookmark folder
4) Bookmark
- try to edit a bookmark
5) Bookmark folder
- try to add a bookmark folder
6) Bookmark
- try to remove a bookmark
7) Bookmark folder
- try to remove a bookmark folder
8) Bookmark
- try to add a bookmark into a folder (with a right click on a folder)
- folder should be preselected
9) Bookmark folder
- try to add a bookmark folder into a folder (with a right click on a folder)
- folder should be preselected
10) Navigation bar
- try to bookmark a page via navigation bar star
11) Navigation bar
- try to edit a bookmark via navigation bar star
12) Bookmark manager
- try to add a bookmark folder from a bookmark manager  (list, folder tree and button)
13) Bookmark manager
- try to add a bookmark from a bookmark manager  (list, folder tree and button)
14) Bookmark manager
- try to edit a bookmark from a bookmark manager (list and folder tree)
15) Bookmark manager
- try to edit a bookmark folder from a bookmark manager (list and folder tree)
16) Bookmark link
- try to bookmark a link (right click on a link in the page)
17) Bookmark manager
- try to dnd bookmark into another folder
18) Bookmark manager
- try to dnd multiple bookmarks into another folder
19) Bookmark manager
- try to dnd folder into another folder
20) Bookmark manager
- try to reorder bookmarks in bookmark manager
21) Bookmark toolbar
- try to dnd bookmark into the folder
22) Bookmark toolbar
- try to dnd folder into another folder
23) Bookmark toolbar
- try to to reorder bookmarks with dnd (try it with folders as well)
24) History manager
- check if history is successfully added
25) History manager
- right click on a history item
- try to delete it
26) Pinned tabs
- try to pin a site
- restart the browser
- check if pinned tab is still there
27) Importer
- try to import HTML file
27) Importer
- try to import from chrome
27) Importer
- try to import from FF
28) Exporter
- check if export is working
29) Url bar suggestions
- check if bookmark is shown in suggestion list
30) Url bar suggestions
- check if history site is shown in suggestion list
31) Ledger
- enabled ledger
- visit some site
- check if site is added to the ledger
32) New tab
- check if top sites are displayed correctly 
33) New tab
- check if you can remove top site
34) New tab
- check if you can pin top site
35) New tab
- check if you can bookmark and edit bookmark already bookmarked top site
36) New tab
- check if you can reorder top sites



Reviewer Checklist:

Tests
